core/src/SYCL/Kokkos_SYCL.cpp

```diff
@@ -93,7 +93,7 @@ bool SYCL::impl_is_initialized() {
 
 void SYCL::impl_finalize() { Impl::SYCLInternal::singleton().finalize(); }
 
-void SYCL::fence() const { m_space_instance->m_queue->wait(); }
+void SYCL::fence() const { m_space_instance->m_queue->wait_and_throw(); }
```

Comment:
Please elaborate on this change with a comment.

Reply:
I am happy to address these comments in #3609. :slightly_smiling_face: